src/Features/Core/Portable/Completion/PreselectionPriority.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    /// <summary>
+    /// An additional hint to the matching algorithm that can
+    /// augment or override the existing text-based matching.
+    /// </summary>
+    internal enum MatchPriority
+    {
+        /// <summary>
+        /// The matching algorithm should give this item no special treatment.
+        /// </summary>
+        Default = 0,
+
+        /// <summary>
+        /// The matching algorithm will tend to prefer this item unless
+        /// a dramatically better text-based match is available or relevant
+        /// items are marked with <see cref="Preselect"/>.
+        /// 
+        /// With no filter text, this item (or the first item alphabeitcally 
+        /// with this priority) should always be selected unless other items
+        /// are marked with <see cref="Preselect"/>.
```

Comment:
Can we document these flags by examples of when you should pick one? I understand that the ambiguity is somewhat by design here (and maybe that's how it needs to be), but it is difficult to get a sense of how "in practice" this all merges together.